x/campaign/types/voucher.go

```diff
@@ -0,0 +1,56 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+const (
+	// VoucherPrefix is the prefix used to represent a voucher denomination
+	VoucherPrefix = "v/"
+)
+
+// SharesToVouchers returns new Coins vouchers from the Shares representation
+func SharesToVouchers(campaignID uint64, shares Shares) sdk.Coins {
+	vouchers := make(sdk.Coins, len(shares))
+	for i, coin := range shares {
+		denom := strings.TrimPrefix(coin.Denom, SharePrefix)
+		coin.Denom = VoucherName(campaignID, denom)
+		vouchers[i] = coin
+	}
+	return vouchers
+}
+
+// VouchersToShares returns new Shares from the Coins vouchers representation
+func VouchersToShares(vouchers sdk.Coins) Shares {
+	shares := make(Shares, len(vouchers))
+	for i, coin := range vouchers {
+		splitDenom := strings.Split(coin.Denom, "/")
```

Comment:
I think it would be proper to have a common method to compute the prefix of the voucher and use `TrimPrefix`